op/model.py

```diff
@@ -302,3 +314,9 @@ def relation_set(self, relation_id, key, value):
 
     def config_get(self):
         return self._run('config-get')
+
+    def pod_spec_get(self):
+        return self._run('pod-spec-get')
+
+    def pod_spec_set(self, pod_spec_data):
+        run(['pod-spec-set'], check=True, input=json.dumps(pod_spec_data).encode('utf8'))
```

Comment:
Why isn't this going through self._run?  That means any custom handling we perform there will need to be duplicated here, and we'll almost certainly forget about it.

Reply:
I didn't use `self._run` because of the need for the `input` param.  However, the latest refactor uses files making that moot, so it's been converted to `self._run_no_output`.